.github/workflows/bazel.yml

```diff
@@ -131,6 +88,17 @@ jobs:
           options: -v ${{ github.workspace }}:/workspaces/magma/
           run: |
             cd /workspaces/magma
+            if [ -n "${{ secrets.BAZEL_REMOTE_PASSWORD }}" ]
```

Comment:
this setup is needed here in two places and we plan to use it in other workflows as well - can this be refactored in a script file with parameters pw and cache key?

Reply:
Added script and refactoring.

See CI runs with the changes:
- [With secret on fork](https://github.com/LKreutzer/magma/actions/runs/2337887288)
- [Without secret on PR](https://github.com/magma/magma/actions/runs/2337876477)